akka-projection-testkit/src/test/scala/akka/projection/testkit/ProjectionTestKitSpec.scala

```diff
@@ -98,25 +98,39 @@ class ProjectionTestKitSpec extends ScalaTestWithActorTestKit with AnyWordSpecLi
 
       val streamFailureMsg = "stream failure"
 
-      val concatenatedRef = new StringBuffer("")
+      val strBuffer = new StringBuffer("")
 
       // this source will 'emit' an exception and fail the stream
       val src = Source.single(1).concat(Source.failed(new RuntimeException(streamFailureMsg)))
 
-      val prj = TestProjection(src, concatenatedRef, _ <= 4)
+      val prj = TestProjection(src, strBuffer, _ <= 4)
 
       val exp =
         intercept[RuntimeException] {
           projectionTestKit.run(prj) {
-            concatenatedRef.toString shouldBe "1-2-3-4"
+            strBuffer.toString shouldBe "1-2-3-4"
           }
         }
 
       exp.getMessage shouldBe streamFailureMsg
     }
+
+    "run a projection with a TestSink" in {
+
+      val strBuffer = new StringBuffer("")
+      val projection = TestProjection(Source(1 to 5), strBuffer, _ <= 5)
+
+      val sinkProbe = projectionTestKit.runWithTestSink(projection)
+
+      sinkProbe.request(5)
+      sinkProbe.expectNextN(5)
+      sinkProbe.expectComplete()
+
+      strBuffer.toString shouldBe "1-2-3-4-5"
```

Comment:
This of course a simple usage of TestSink, but we can imagine the more elaborated cases. 

Giving the users full control over the flow will be very useful.

Reply:
I think exposing internals a bit to give users access to the underlying stream will make sense in testing scenarios.  I think it might also make sense in non-testing scenarios, but I'll think about that some more and get back to you.